### ISSUE TITLE
[3.2] Matrix Propagation Issues with Multi-Site

### ISSUE THREAD

**Issue by caseyreid** [craftcms/cms#4698]
### Description

We're working on setting up a new language via Craft's multi-site functionality. This site has multiple matrix fields with content that will need to be translated to the new language.

Prior to 3.2, we could set the Matrix translation setting to "Manage blocks on a per-site basis". If we then enabled the new site for one of our sections, with the setting to "Propagate entries across all enabled sites?", our entries and the matrix data from the original site would copy over to the new site and provide us with a starting point for our content. After that, we could make changes to the matrix content in the new site, without affecting the original matrix content on the original site. This was great for providing a clone of the content as a starting point, from which to then edit from.

Since 3.2, this scenario does not seem to be possible. If we set the Matrix field to "Only save blocks to the site they were created in", then when the entries first get propagated to the new site, none of the Matrix data is copied over as a starting point. If we instead set the Matrix field to "Save blocks to all sites the owner element is saved in", then it does copy the Matrix data over to the new site when entries are first propagated, however, if we then edit any of the Matrix content on the new site after that, it edits it for both sites.

Changing the setting after the entry propagation also does not help. For example, I thought I might be able to first set the Matrix field to "Save blocks to all sites the owner element is saved in", so that the data was originally copied over to the new site entries during propagation. Then after propagation, change the Matrix setting to  "Manage blocks on a per-site basis", so I could manage the original content on a per site basis moving forward. Changing that Matrix setting after entry propagation actually still causes the same issue. Which is that it erases all the originally cloned Matrix data from the new site, so that it can be managed on a per site basis.

Aside from making it difficult to clone a site as a starting point, this Matrix setting behavior also seems kind of dangerous in that Matrix data originally added to a new site during propagation can be deleted from that site if you change the Matrix field settings.

### Steps to reproduce

1. Set Matrix field to "Save blocks to all sites the owner element is saved in".
2. Enabled a section for a new site with the setting set to "save entries to all sites enabled for this section".
3. After entries have propagated, and matrix content has been copied to the new entries from the original site, update the matrix field setting to "Manage blocks on a per-site basis".
4. This causes the originally copied Matrix data to be removed from the new site entry.

### Additional info

- Craft version: 3.2.7
- PHP version: 7.1.26
- Database driver & version: MySQL 5.7.25

**Comment by brandonkelly** [comment]
Just fixed this for the next release. Thanks for reporting!

**Comment by elfacht** [comment]
@brandonkelly This doesn't seem to be fixed in `3.2.9` or will it ship with the next release?

**Comment by brandonkelly** [comment]
@elfacht This issue was specifically about what happens the moment you enable a section for a new site.

> Once you choose 'same group' will delete all the blocks. If you switch back to 'per-site' they're still gone.

Had you done that before or after enabling the section for a new site? If after, the current behavior is that blocks get resaved (from the current site if they exist there, but otherwise it's a crapshoot which site they get queried in), and in the process they will be removed from any sites that aren't in the same group as the site they were queried in. They wouldn't end up going completely missing though.

**Comment by elfacht** [comment]
@brandonkelly It was basically first **Save blocks to all sites the owner element is saved in** on content production until launch. Then I switched to **Save blocks to other sites in the same site group** and on production everything was messed up and different from local env. Like every block was duplicated and stuff, other groups even got four times the blocks. Pretty weird.

I tried then **Only save blocks to the site they were created in** on local today with a fresh live DB dump and it worked for once when I opened the entry. When I saved it, all other sites has lost their blocks. It feels a bit fragile to change it but sticking with **Save blocks to all sites the owner element is saved in** forever doesn't feel like a solution.

>  If after, the current behavior is that blocks get resaved (from the current site if they exist there, but otherwise it's a crapshoot which site they get queried in)

I waited for it to complete, but still it didn't work. What would be the best workflow? Right now local and live works with **Save blocks to other sites with the same language**, but it would be perfect if I could easily switch to either of the both options.
